src/main/typescript/rug/operations/Decorators.ts

```diff
@@ -27,137 +30,190 @@ function get_metadata(obj: any, key: string) {
 /**
  * Decorator for parameters. Adds to object properties
  */
-function Parameter(details: BaseParameter) {
+export function Parameter(details: BaseParameter) {
   return (target: any, propertyKey: string) => {
-    let params = get_metadata(target, "__parameters");
-    if (params == null) {
-      params = [];
-    }
-    const copy: any = { ...details };
-    copy.name = propertyKey;
-    copy.decorated = true;
-    params.push(copy);
-    set_metadata(target, "__parameters", params);
+    declareParameter(target, propertyKey, details);
   };
 }
+
+export function declareParameter(target: any, propertyKey: string, details: BaseParameter) {
```

Comment:
Is there no way to have IntelliJ IDEA use a four-space indent for TypeScript files?